src/deploy/functions/release.ts

```diff
@@ -60,26 +60,21 @@ export async function release(context: args.Context, options: args.Options, payl
 
   // Note(inlined): We might increase consistency if we tried a fully regional strategy, but
   // the existing code was written to process deletes before creates and updates.
-  const allFnsToDelete = Object.values(fullDeployment.regionalDeployments)
-    .map((regionalChanges) => regionalChanges.functionsToDelete)
-    .reduce((accum, functions) => [...(accum || []), ...functions]);
+  const allFnsToDelete = Object.values(fullDeployment.regionalDeployments).reduce(
+    (accum, region) => [...accum, ...region.functionsToDelete],
+    [] as backend.FunctionSpec[]
+  );
   const shouldDeleteFunctions = await promptForFunctionDeletion(
     allFnsToDelete,
     options.force,
     options.nonInteractive
   );
-  if (shouldDeleteFunctions) {
```

Comment:
It looks like this new code will always delete functions, even shouldDeleteFunctons = false.

Reply:
@#$! it removes the topics and schedules from the delete list and still deletes the function. Will fix.

I guess the most dangerous PR is the one that's "too trivial to mess up"